src/components/SubnavigationButton/SubnavigationButton.css

```
@@ -35,69 +29,93 @@
 .SubnavigationButton__expandableIcon {
   margin-left: 8px;
   margin-top: 1px; /* Смещает иконку относительно текста */
-  color: var(--icon_secondary);
+  color: var(--icon_secondary, var(--vkui--color_icon_secondary));
 }
 
-/**
- * selected
- */
-
-.SubnavigationButton--selected {
-  background-color: var(--button_primary_background);
-  color: var(--button_primary_foreground);
+.SubnavigationButton__in > .SubnavigationButton__after:first-child,
+.SubnavigationButton__in > .SubnavigationButton__label:first-child {
+  padding-left: var(
+    --vkui--size_button_base_small_padding_horizontal_icon--regular
+  );
 }
 
-.SubnavigationButton--selected .SubnavigationButton__before {
-  color: inherit;
+.SubnavigationButton__in > .SubnavigationButton__label:last-child {
+  padding-right: var(
+    --vkui--size_button_base_small_padding_horizontal_icon--regular
+  );
 }
 
-.SubnavigationButton--selected .Counter {
-  background-color: var(--button_primary_foreground);
-  color: var(--button_primary_background);
+/**
+ * mode
+ */
+
+.SubnavigationButton--mode-primary {
+  background-color: var(
+    --modal_card_background,
+    var(--vkui--color_background_modal)
+  );
+  box-shadow: var(--vkui--elevation3);
 }
 
-.SubnavigationButton--selected .SubnavigationButton__expandableIcon {
-  color: inherit;
+.SubnavigationButton--mode-outline {
+  box-shadow: inset 0 0 0 var(--thin-border)
+    var(--vkui--color_field_border_alpha);
 }
 
 /**
- * size m
+ * selected
  */
 
-.SubnavigationButton--m .SubnavigationButton__in {
-  height: 36px;
-  padding: 0 14px;
+.SubnavigationButton--selected {
+  background-color: var(
+    --button_primary_background,
+    var(--vkui--color_background_accent)
+  );
+  color: var(--button_primary_foreground, var(--vkui--color_text_contrast));
+  box-shadow: none;
+  border: none;
 }
 
-/**
- * size l
- */
+.SubnavigationButton--selected.SubnavigationButton--mode-tertiary {
+  background-color: var(--vkui--color_transparent--active);
+  color: var(--vkui--color_text_primary);
+}
 
-.SubnavigationButton--l .SubnavigationButton__in {
-  height: 44px;
-  padding: 0 14px;
+.SubnavigationButton--selected:not(.SubnavigationButton--mode-tertiary)
+  .SubnavigationButton__before {
+  color: inherit;
+}
+
+.SubnavigationButton--selected:not(.SubnavigationButton--mode-tertiary)
+  .SubnavigationButton__expandableIcon {
+  color: inherit;
+}
+
+.SubnavigationButton--selected:not(.SubnavigationButton--mode-tertiary).SubnavigationButton--mode-tertiary
+  .SubnavigationButton__expandableIcon {
+  color: var(--vkui--color_icon_accent);
```

Думаю, тут стоит опереться на силу каскада. 😉 

```suggestion
.SubnavigationButton--selected {
  background-color: var(
    --button_primary_background,
    var(--vkui--color_background_accent)
  );
  color: var(--button_primary_foreground, var(--vkui--color_text_contrast));
  box-shadow: none;
  border: none;
}

.SubnavigationButton--selected .SubnavigationButton__before,
.SubnavigationButton--selected .SubnavigationButton__expandableIcon {
  color: inherit;
}

.SubnavigationButton--selected.SubnavigationButton--mode-tertiary {
  background-color: var(--vkui--color_transparent--active);
  color: var(--vkui--color_text_primary);
}

.SubnavigationButton--selected.SubnavigationButton--mode-tertiary 
.SubnavigationButton__expandableIcon {
  color: var(--vkui--color_icon_accent);
```

---

✅